### PR TITLE
emit power optimization status, Fixes AB#3080614

### PR DESCRIPTION
**Pull request by rpdome**
[AB#3080614](https://identitydivision.visualstudio.com/fac9d424-53d2-45c0-91b5-ef6ba7a6bf26/_workitems/edit/3080614)

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/adal/internal/PowerManagerWrapper.java [hunk 1]
```diff
@@ -67,6 +68,58 @@ public boolean isDeviceIdleMode(final Context connectionContext) {
...
+            final PowerManager powerManager = ((PowerManager) context.getSystemService(Context.POWER_SERVICE));
+            if (powerManager.isDeviceIdleMode()) {
+                return "Idle";
+            }
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU &&
+                    powerManager.isDeviceLightIdleMode()) {
+                return "LightIdle";
+            }
+        } catch (Exception e){
```
Comment (p3dr0rv) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/PowerManagerWrapper.java]: nit: final

File: common/src/main/java/com/microsoft/identity/common/adal/internal/PowerManagerWrapper.java [hunk 2]
```diff
@@ -67,6 +68,58 @@ public boolean isDeviceIdleMode(final Context connectionContext) {
...
+            }
+
+            final PowerManager powerManager = ((PowerManager) context.getSystemService(Context.POWER_SERVICE));
+            if (powerManager.isIgnoringBatteryOptimizations(context.getPackageName())){
+                return "OptOut";
+            } else {
+                return "";
+            }
+
+        } catch (Exception e){
```
Comment (p3dr0rv) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/PowerManagerWrapper.java]: nit: final

File: common/src/main/java/com/microsoft/identity/common/adal/internal/PowerManagerWrapper.java
Comment (p3dr0rv) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/PowerManagerWrapper.java:1]: nit: run Code formatter

**Review by p3dr0rv — APPROVED** [review]

File: common/src/main/java/com/microsoft/identity/common/adal/internal/PowerManagerWrapper.java [hunk 3]
```diff
@@ -67,6 +68,58 @@ public boolean isDeviceIdleMode(final Context connectionContext) {
         return ((PowerManager) connectionContext.getSystemService(Context.POWER_SERVICE)).isDeviceIdleMode();
     }
 
+    /**
+     * Gets a string representing Device Idle status.
+     * Will return an empty string if the device is not in any idle mode.
+     * (Possible Values: "Idle", "LightIdle", "Unknown" , "")
+     */
+    public String getDeviceIdleMode(final Context context){
```
Comment (mohitc1) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/PowerManagerWrapper.java]: >final Context context [](http://example.com/codeflow?start=36&length=21)

nit: NonNull

**Review by mohitc1 — APPROVED** [review]
:shipit:

**Pull request merged**